control.go

```diff
@@ -257,16 +257,22 @@ func DecodeControl(packet *ber.Packet) Control {
 	ControlType := packet.Children[0].Value.(string)
 	Criticality := false
 
+	var value *ber.Packet
 	packet.Children[0].Description = "Control Type (" + ControlTypeMap[ControlType] + ")"
-	value := packet.Children[1]
-	if len(packet.Children) == 3 {
-		value = packet.Children[2]
-		packet.Children[1].Description = "Criticality"
-		Criticality = packet.Children[1].Value.(bool)
+	if len(packet.Children) > 1 {
+		value = packet.Children[1]
+		if len(packet.Children) == 3 {
+			value = packet.Children[2]
+			packet.Children[1].Description = "Criticality"
+			Criticality = packet.Children[1].Value.(bool)
```

Comment:
I went ahead and did the switch check in https://github.com/go-ldap/ldap/pull/83/